src/service/DataReader.java

```diff
@@ -0,0 +1,185 @@
+package service;
+
+import address.Address;
+import food.*;
+import order.Order;
+import restaurant.Restaurant;
+import user.*;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.UUID;
+
+public class DataReader {
+    private static DataReader reader;
+    public static DataReader getInstance() {
+        if(reader == null) {
+            reader = new DataReader();
+        }
+        return reader;
+    }
```

Comment:
private constructor missing, so your class is not singleton :)  same as for the writer class